lib/game_team.rb

```diff
@@ -257,24 +257,29 @@ def self.opponents_wins(team_id)
     opponent_wins
   end
 
-  #THIS ONE FAILS AND CUASES RSPEC TO FREEZE UP
+####1:15 seconds
   def self.favorite_opponent_id(team_id)
+    team_id = team_id.to_i
     record_length = {}
     opponents_records(team_id).map do |team_id, record|
       record_length[team_id] = record.length
     end
     opponents = opponents_records(team_id).keys
     opponents.min_by {|opponent| (opponents_wins(team_id)[opponent].to_f / record_length[opponent].to_f).round(2)}
   end
-  #THIS ONE FAILS AND CUASES RSPEC TO FREEZE UP
+####2:30 seconds
   def self.rival_id(team_id)
+    team_id = team_id.to_i
     record_length = {}
     opponents_records(team_id).map do |team_id, record|
       record_length[team_id] = record.length
     end
     opponents = opponents_records(team_id).keys
     opponents.max_by {|opponent| (opponents_wins(team_id)[opponent].to_f / record_length[opponent].to_f).round(2)}
   end
+#
```

Comment:
does not freeze. does take a long time. noted. I wonder why this takes twice as long as the method above which is nearly identical.